BlueprintUI/Sources/Layout/Stack.swift

```diff
@@ -537,9 +591,18 @@ extension StackLayout {
 
         var axisOrigin: CGFloat = 0.0
 
-        let axisSegments = zip(basisSizes, shrinkPriorities).map { basis, shrinkPriority -> Segment in
+        let axisSegments = basisSizes.enumerated().map { index, basis -> Segment in
```

Comment:
Fwiw, might be worth adding and using an `indexedMap` here given its a pretty hot codepath. When I was doing performance testing; I saw (even in release mode) about a 25% perf bump from using this:

```
extension Array {

    /// A `map` implementation that also passes the `index` of each element in the original array.
    ///
    /// This method is more performant than calling `array.enumerated().map(...)` by up
    /// to 25% for large collections, so prefer it when needing an indexed `map` in areas where performance is critical.
    @inlinable func indexedMap<Mapped>(_ map : (Int, Element) -> Mapped) -> [Mapped] {

        let count = self.count

        var mapped = [Mapped]()
        mapped.reserveCapacity(count)

        for index in 0..<count {
            mapped.append(map(index, self[index]))
        }

        return mapped
    }
```

I believe because there's no enumeration structure being created.